.github/workflows/build.yml

```diff
@@ -8,70 +8,71 @@ on:
-
-    strategy:
-      matrix:
-        node-version: ["12", "14", "16"]
-
     steps:
-      - uses: actions/checkout@v2
-      - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v2
+      - uses: actions/checkout@v3
+
+      - name: Use Node.js ${{ env.node-version }}
+        uses: actions/setup-node@v3
         with:
-          node-version: ${{ matrix.node-version }}
-      - run: yarn install --frozen-lockfile
-      - run: yarn run build
-      #- run: yarn run test TODO: Readd once we have tests
+          node-version: ${{ env.node-version }}
+
+      - name: Install Dependencies
+        run: yarn install --immutable
+
+      - name: Build
+        run: yarn run build
+
+      #- run: yarn run test # Uncomment and run tests here when implemented in package.json
 
   lint:
     runs-on: ubuntu-latest
-
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+
       - name: Use Node.js ${{ env.node-version }}
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v3
         with:
           node-version: ${{ env.node-version }}
-      - run: yarn install --frozen-lockfile
-      - run: yarn run lint
 
-  build:
+      - name: Install Dependencies
+        run: yarn install --immutable
+
+      - name: ESLint & Prettier
+        run: yarn run lint
+
+  push:
     runs-on: ubuntu-latest
     needs: [test, lint]
-    if: github.ref == 'refs/heads/main'
-
+    if: github.ref == 'refs/heads/main' # Only runs on commits to the main branch
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+
       - name: Use Node.js ${{ env.node-version }}
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@v3
         with:
           node-version: ${{ env.node-version }}
-      - run: |
-          git config user.name "O.A.F."
-          git config user.email "<>"
-      - run: |
-          cp -R mafia mafiafiles
-      - run: yarn install --frozen-lockfile
-      - run: yarn run build
-      - run: |
-          git fetch --all
-          git switch release
-          rm -r scripts/
-          rm -r data/
-          mv KoLmafia/scripts scripts
-          mv mafiafiles/data ./data
-      - run: |
-          git add scripts/
-          git add data/
-          {
-            git commit -m "${{github.event.commits[0].message}}"
-            git push origin release
-          } || exit 0
+
+      - name: Install Dependencies
+        run: yarn install --immutable
+
+      - name: Build
+        run: yarn run build
+
+      - name: Push to Release
+        uses: s0/git-publish-subdir-action@develop
+        env:
+          REPO: self
+          BRANCH: release
+          FOLDER: ${{ env.path }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # GitHub will automatically add this - you don't need to bother getting a token
+          MESSAGE: "Build: ({sha}) {msg}" # The commit hash and message that prompted this workflow
```

Comment:
I think the commit hash is probably just going to clutter things up, but if you disagree, I am not particularly invested in that opinion.

Reply:
https://github.com/loathers/greenbox/tree/release

Here's an example of what it looks like.

Reply:
Oh it's the short version

Reply:
Yeah that' scool as hell